core/src/main/java/io/grpc/internal/DelayedStream.java

```diff
@@ -63,6 +63,7 @@ class DelayedStream implements ClientStream {
   private long streamSetTimeNanos;
   // No need to synchronize; start() synchronization provides a happens-before
   private List<Runnable> preStartPendingCalls = new ArrayList<>();
+  protected Status lastPickStatus;
```

Comment:
Put this in `PendingStream`, as it is specific to wait-for-ready which is specific to DelayedClientTransport. Also, let's only show it when the RPC is wait-for-ready, as that's the only case that there could be an error status and it wasn't used. In fact, if the status isn't an error, let's not show it for wait-for-ready either.